synapse/handlers/federation.py

```diff
@@ -2164,15 +2176,23 @@ def _update_auth_events_and_context_for_auth(
                     event.auth_event_ids()
                 )
             except Exception:
-                # FIXME:
                 logger.exception("Failed to get auth chain")
 
         if event.internal_metadata.is_outlier():
+            # XXX: given that, for an outlier, we'll be working with the
+            # event's *claimed* auth events rather than those we calculated:
+            # (a) is there any point in this test, since different_auth below will
+            # obviously be empty
+            # (b) alternatively, why don't we do it earlier?
             logger.info("Skipping auth_event fetch for outlier")
             return context
 
         # FIXME: Assumes we have and stored all the state for all the
         # prev_events
+        #
+        # FIXME: what does the fixme above mean? where do prev_events come into
```

Comment:
I feel like we probably should resolve this before merging the PR in order to avoid having `FIXME` comments referring to other `FIXME` comments.

Reply:
The `FIXME` is mostly just saying that we are assuming that we've actually persisted everything that we go on to look up in the DB. Broadly I think this is just lamenting a broader point that if we get a bunch of events in a room we persist them one by one, rather than handling them as a chunk and persisting them all at once. IYSWIM

Reply:
as per #synapse-dev: erik says burn the FIXME